### PR TITLE
Explicitly encode if an entity should be included in the picker buffer

### PR DESCRIPTION
**Pull request by kokosha**
After sorting the entities from back to front, the logic from picker ignoring first two elements need to be changed.

### CODE REVIEW

File: korangar/src/graphics/passes/picker/entity.rs [hunk 1]
```diff
@@ -214,6 +214,7 @@ impl Drawer<{ BindGroupCount::One }, { ColorAttachmentCount::One }, { DepthAttac
 
 impl Prepare for PickerEntityDrawer {
     fn prepare(&mut self, device: &Device, instructions: &RenderInstruction) {
+        // TODO: Pass the player size directly instead of hardcoding it here.
```
Comment (hasenbanck) [inline korangar/src/graphics/passes/picker/entity.rs:217]:
```rust
fn prepare(&mut self, device: &Device, instructions: &RenderInstruction) {
        self.instance_data.clear();
        self.draw_count = 0;

        if self.bindless_support {
            self.bump.reset();
            self.lookup.clear();

            let mut texture_views = Vec::with_capacity_in(self.draw_count, &self.bump);

            for instruction in instructions.entities.iter().filter(|instruction| instruction.pickerable) {
                self.draw_count += 1;
```

Let's simplify this like this.

File: korangar/src/graphics/passes/picker/entity.rs [hunk 2]
```diff
@@ -264,10 +262,7 @@ impl Prepare for PickerEntityDrawer {
             self.instance_data_buffer.reserve(device, self.instance_data.len());
             self.bind_group = Self::create_bind_group_bindless(device, &self.bind_group_layout, &self.instance_data_buffer, &texture_views)
         } else {
-            // We skip the first entity, because we don't want the player entity to show up
-            // in the picker buffer.
-            // TODO: Remove the player entity correctly.
-            for instruction in instructions.entities.iter().skip(player_size) {
+            for instruction in instructions.entities.iter().filter(|instruction| instruction.pickerable == true) {
```
Comment (hasenbanck) [inline korangar/src/graphics/passes/picker/entity.rs:265]:
```rust
            for instruction in instructions.entities.iter().filter(|instruction| instruction.pickerable) {
                self.draw_count += 1;
```

Like above.

File: korangar/src/graphics/passes/picker/entity.rs [hunk 3]
```diff
@@ -229,10 +230,7 @@ impl Prepare for PickerEntityDrawer {
 
             let mut texture_views = Vec::with_capacity_in(self.draw_count, &self.bump);
```
Comment (hasenbanck) [inline korangar/src/graphics/passes/picker/entity.rs:231]:
```suggestion
            let mut texture_views = Vec::with_capacity_in(instructions.entities.len(), &self.bump);
```

File: korangar/src/graphics/instruction.rs [hunk 4]
```diff
@@ -185,6 +185,7 @@ pub struct EntityInstruction {
     pub color: Color,
     pub mirror: bool,
     pub entity_id: EntityId,
+    pub pickerable: bool,
```
Comment (hasenbanck) [inline korangar/src/graphics/instruction.rs:188]: Let's call this `add_to_picker`.

File: korangar/src/world/entity/mod.rs [hunk 5]
```diff
@@ -1130,8 +1131,8 @@ impl Entity {
             .generate_pathing_mesh(device, queue, map, pathing_texture_mapping);
     }
 
-    pub fn render(&self, instructions: &mut Vec<EntityInstruction>, camera: &dyn Camera) {
-        self.get_common().render(instructions, camera);
+    pub fn render(&self, instructions: &mut Vec<EntityInstruction>, camera: &dyn Camera, pickerable: bool) {
```
Comment (hasenbanck) [inline korangar/src/world/entity/mod.rs:1134]: Let's also call this `add_to_picker`

File: korangar/src/world/map/mod.rs [hunk 6]
```diff
@@ -265,10 +265,11 @@ impl Map {
 
     #[cfg_attr(feature = "debug", korangar_debug::profile)]
     pub fn render_entities(&self, instructions: &mut Vec<EntityInstruction>, entities: &[Entity], camera: &dyn Camera, include_self: bool) {
-        entities
-            .iter()
-            .skip(!include_self as usize)
-            .for_each(|entity| entity.render(instructions, camera));
+        let mut is_current_player = true || include_self;
```
Comment (hasenbanck) [inline korangar/src/world/map/mod.rs:268]:
```rust
    #[cfg_attr(feature = "debug", korangar_debug::profile)]
    pub fn render_entities(&self, instructions: &mut Vec<EntityInstruction>, entities: &[Entity], camera: &dyn Camera) {
        entities.iter().enumerate().for_each(|(index, entity)| {
            entity.render(instructions, camera, index != 0);
        });
    }
```

We never call this function with "include_self==false"! So let's simplify it.

**Review by hasenbanck — CHANGES_REQUESTED** [review]
Can you also please update the commit message?

`Explicitly encode if an entity should be included in the picker buffer`

File: korangar/src/graphics/passes/picker/entity.rs [hunk 7]
```diff
@@ -199,7 +199,8 @@ impl Drawer<{ BindGroupCount::One }, { ColorAttachmentCount::One }, { DepthAttac
             let mut current_texture_id = self.solid_pixel_texture.get_id();
             pass.set_bind_group(2, self.solid_pixel_texture.get_bind_group(), &[]);
 
-            for (index, instruction) in draw_data[1..self.draw_count.saturating_add(1)].iter().enumerate() {
+            let instructions = draw_data.iter().filter(|instruction| instruction.add_to_picker);
+            for (index, instruction) in instructions.enumerate() {
```
Comment (kokosha) [inline korangar/src/graphics/passes/picker/entity.rs:203]: Find a bug in non-bind mode and changed to correct the behavior

**Review by hasenbanck — APPROVED** [review]

**Pull request merged**